### ISSUE TITLE
[Ubuntu 20.04 / Gnome 3.36.1]: Interval daemon dies after first change (fix included)

### ISSUE THREAD

**Issue by eode** [BigE/desk-changer#68]
Thanks for making this extension!

Here's the traceback from syslog:

```
Apr 24 17:30:47 hostname gnome-shell[63722]: JS ERROR: Error: Wrong type undefined; string expected
_emit_preview@/home/username/.local/share/gnome-shell/extensions/desk-changer@eric.gach.gmail.com/daemon/profile.js:216:14
fill_queue@/home/username/.local/share/gnome-shell/extensions/desk-changer@eric.gach.gmail.com/daemon/profile.js:102:18
next@/home/username/.local/share/gnome-shell/extensions/desk-changer@eric.gach.gmail.com/daemon/profile.js:163:14
next@/home/username/.local/share/gnome-shell/extensions/desk-changer@eric.gach.gmail.com/daemon/server.js:189:73
__callback__@/home/username/.local/share/gnome-shell/extensions/desk-changer@eric.gach.gmail.com/daemon/timer.js:39:33
```

**Comment by eode** [comment]
This can be fixed by editing the following:

Line 216, in `daemon/profile.js`, in `_emit_preview()`, change:
`        this.emit('preview', wallpaper);`
to
`        this.emit('preview', wallpaper ? wallpaper : '');`

There may be a better place to change this (what is the source of the `undefined?`), but this works for me for now, and hopefully this saves you a smidge of work.

**Comment by BigE** [comment]
Thanks for reporting, but I don't think that's the original issue. If you look at your stack trace and find https://github.com/BigE/desk-changer/blob/474865b210a930609b7ca6d04334f2354ace660f/desk-changer%40eric.gach.gmail.com/daemon/profile.js#L102 this bit of code, it's pulling from a queue object that already has data. If this is "first run" that means the profile state in gsettings is bad. 

What's baffling me a little bit is where the stack starts from... https://github.com/BigE/desk-changer/blob/master/desk-changer%40eric.gach.gmail.com/daemon/server.js#L189 this is the lockscreen that's been... "changed"... in 3.36. I did some work earlier to remove it from the extension https://github.com/BigE/desk-changer/commit/4587c206a04af49b35b27177a80e751c54a96f43 but it appears that I might have further work to do into undoing that.

Unfortunately, while your fix gets you running, it doesn't fix the larger problem I've created here. I'll do more digging. Thanks again!
